src/Module.cpp

```diff
@@ -227,30 +227,7 @@ Module link_modules(const std::string &name, const std::vector<Module> &modules)
 
 Buffer<uint8_t> Module::compile_to_buffer() const {
     // TODO: This Hexagon specific code should be removed as soon as possible.
-    bool use_shared_object = target().has_feature(Target::HVX_shared_object);
-
-    if (use_shared_object) {
-        return compile_module_to_hexagon_shared_object(*this);
-    } else {
-        llvm::LLVMContext context;
-        std::unique_ptr<llvm::Module> llvm_module(compile_module_to_llvm_module(*this, context));
-
-        llvm::SmallVector<char, 4096> object;
-        llvm::raw_svector_ostream object_stream(object);
-        compile_llvm_module_to_object(*llvm_module, object_stream);
-
-        if (debug::debug_level() >= 2) {
-            debug(2) << "Submodule assembly for " << name() << ": " << "\n";
-            llvm::SmallString<4096> assembly;
-            llvm::raw_svector_ostream assembly_stream(assembly);
-            compile_llvm_module_to_assembly(*llvm_module, assembly_stream);
-            debug(2) << assembly.c_str() << "\n";
-        }
-
-        Buffer<uint8_t> result(object.size(), name());
-        memcpy(result.data(), reinterpret_cast<uint8_t*>(&object[0]), object.size());
-        return result;
-    }
+    return compile_module_to_hexagon_shared_object(*this);
```

Comment:
@zvookin you might want to look at this in particular. I think we need "compile_to_shared_object" to be more general (not Hexagon specific), and a mechanism for submodules to specify how they want to be compiled. For another PR...